src/main/java/spoon/reflect/visitor/chain/CtQuery.java

```diff
@@ -16,56 +16,124 @@
  */
 package spoon.reflect.visitor.chain;
 
+import spoon.reflect.declaration.CtElement;
 import spoon.reflect.visitor.Filter;
 
 import java.util.List;
 
 /**
  * CtQuery represents a query, which can be used to traverse a spoon model and collect
- * children elements in several ways.<br>
+ * children elements in several ways.<br/>
  *
- * <br>
- * Use {@link CtQueryable#map(CtFunction)}} or {@link CtQueryable#filterChildren(Filter)} to create a new query starting from an existing element.<br>
+ * Creation: A query is created either from a {@link CtElement}, or it can be defined first from {@link CtQueryImpl} and bound to root elements
+ * afterwards using {@link CtQuery#setInput(Object...)}.<br/>
  *
- * The main methods are:
- * <ul>
- * <li> {@link CtQueryable#map(CtFunction)} - uses a lambda expression to return any model elements that are directly accessible from an input element.
- * <li> {@link CtQueryable#filterChildren(Filter)} - uses {@link Filter} instances to filter children of an element
- * <li> {@link #list()} - to evaluate the query and return a list of elements produced by this query.
- * </ul>
- * The query can be used several times.<br>
- * A CtQuery  is lazily evaluated once {{@link #list()}} is called.
- * Usually a new query is created each time when one needs to query something.
- * However, reusing a {@link CtQuery} instance makes sense when the same query has to be evaluated
- * several times in a loop.
+ * Chaining: In a query several steps can be chained, by chaining calls to map functions. The non-null outputs of one step
+ * are given as input to the next step. An iterable or array output is considered as a set of different inputs for the next step.
+ *
+ * Evaluation: A CtQuery is lazily evaluated once {@link CtQuery#list()} or {@link CtQuery#forEach(CtConsumer)} are called.<br/>
  *
- * @param <O> the type of the element produced by this query
  */
-public interface CtQuery<O> extends CtQueryable {
+public interface CtQuery extends CtQueryable {
+
+	/**
+	 * Recursively scans all children elements of an input element.
+	 * The matched child element for which (filter.matches(element)==true) are sent to the next query step.
+	 * Essentially the same as {@link CtElement#getElements(Filter)} but more powerful, because it

```

Comment:
Just to be sure here: when you say filtered elements are sent to the next query step, it's not mandatory: you can just get the list of filtered elements through `list()` right? The meaning of your comment here is that the filterChildren can be used in the chain as it returns a CtQuery?